### PR TITLE
docs(sample): Added AuthorizeDataset Sample

### PR DESCRIPTION
**Pull request by prash-mi**
Added Sample for AuthorizeDataset

### CODE REVIEW

File: samples/snippets/src/main/java/com/example/bigquery/AuthorizeDataset.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
+/*
```
Comment (tswast) [inline samples/snippets/src/main/java/com/example/bigquery/AuthorizeDataset.java:1]: Samples require integration tests to ensure they work. See: https://github.com/googleapis/java-bigquery/tree/main/samples/snippets/src/test/java/com/example/bigquery

**Review by tswast — CHANGES_REQUESTED** [review]

**Review by stephaniewang526 — CHANGES_REQUESTED** [review]
1) Let's put "create a dataset" prep for `userDataset` into the IT's `setUp` method. We can use `BIGQUERY_DATASET_NAME` env var as the `sourceDataset` in our sample (instead of creating another `sourceDataset`).
2) We can pass `sourceDataset` and `userDataset` into the sample method so it's clear that we are trying to authorize userDataset with sourceDataset ACL.
3) We need to delete the created `userDataset` in `cleanUp` method of the IT.

The sample body should just show only the steps to create `DatasetAclEntity` and calling dataset.update API with the new Acl list.

File: samples/snippets/src/main/java/com/example/bigquery/AuthorizeDataset.java (hunk 1 above)
Comment (prash-mi) [reply samples/snippets/src/main/java/com/example/bigquery/AuthorizeDataset.java:1]: sure, added AuthorizeDatasetIT

**Comment by prash-mi** [comment]
> 1. Let's put "create a dataset" prep for `userDataset` into the IT's `setUp` method. We can use `BIGQUERY_DATASET_NAME` env var as the `sourceDataset` in our sample (instead of creating another `sourceDataset`).
> 2. We can pass `sourceDataset` and `userDataset` into the sample method so it's clear that we are trying to authorize userDataset with sourceDataset ACL.
> 3. We need to delete the created `userDataset` in `cleanUp` method of the IT.
> 
> The sample body should just show only the steps to create `DatasetAclEntity` and calling dataset.update API with the new Acl list.

Updated the sample. Now passing the `datasetId` to the sample method and calling the update API

**Comment by stephaniewang526** [comment]
Thanks! 

But this is now blocked by a backend issue: https://b.corp.google.com/issues/211633697#comment39

**Review by stephaniewang526 — APPROVED** [review]

**Comment by prash-mi** [comment]
Closing. I will raise another PR with the required changes

**Pull request closed without merging**